### PR TITLE
[libpq] Fix version mismatch between CONTROL and portfile

### PR DESCRIPTION
**Pull request by AlvinZhangH**
Related issue #9235.
1. Update the CONTROL version
2. Delete deprecated functions
3. Update deprecated functions

No features need to test.

### CODE REVIEW

File: ports/libpq/portfile.cmake [hunk 1]
```diff
@@ -1,5 +1,3 @@
-include(vcpkg_common_functions)
-
 if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
```
Comment (JackBoosY) [inline ports/libpq/portfile.cmake]: Could you change this condition to `if (VCPKG_TARGET_IS_UWP)` ?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by dan-shaw — APPROVED** [review]

**Comment by CJCombrink** [comment]
Related to the discussion on the issue, can I maybe suggest something for the portfile.cmake.
This depends on if it is possible to get the Version set in the CONTROL file from the porfile.
My suggestion:
```
# Say we have something like the following coming in
# set(PORT_VERSION "9.6.3-1")

# Get the base version:
string(REGEX MATCH  "[0-9]+\.[0-9]+\.[0-9]+"  PORT_VERSION_BASE "${PORT_VERSION}")

vcpkg_download_distfile(ARCHIVE
    URLS "https://ftp.postgresql.org/pub/source/v${PORT_VERSION_BASE}/postgresql-${PORT_VERSION_BASE}.tar.bz2"
    FILENAME "postgresql-${PORT_VERSION_BASE}.tar.bz2"
    SHA512 97141972e154e6b0e756ee6a4e20f26e82022a9fd4c56305314a3a5567a79ece638e4ac3d513b46138737ae6bd27a098f30013a94767db151181aac9c01290a1
)
```

Even if you don't have access to such a variable, adding it to the portfile manually will at least make sure the version numbers in the portfile does not get out of sync and adds a single place to update the value.

**Comment by CJCombrink** [comment]
Since it seems like one does not get access to the version from the portifle, the following could be used until such a feature is in place:
```
file(STRINGS "${CURRENT_PORT_DIR}/CONTROL" PORT_VERSION REGEX "Version: .+")
string(REGEX REPLACE  "Version: (.+)" "\\1"  PORT_VERSION "${PORT_VERSION}")
```

Then the code in my previous comment can be used to drop the `-{n}` if there is one

**Comment by JackBoosY** [comment]
@CJCombrink I think this should be called as a vcpkg detection function before download source.

**Pull request merged**